processor/spanmetricsprocessor/cache/cache.go

```diff
@@ -0,0 +1,70 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanmetricsprocessor/cache"
+
+import (
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// Cache consists of an LRU cache and the evicted items from the LRU cache.
+// This data structure makes sure all the cached items can be retrieved either from the LRU cache or the evictedItems
+// map. In spanmetricsprocessor's use case, we need to hold all the items during the current processing step for
+// building the metrics. The evicted items can/should be safely removed once the metrics are built from the current
+// batch of spans.
+type Cache struct {
```

Comment:
This zero value type is no longer safe to use, I ask that you have an interface in its place and move this to be non exported.

Reply:
IMO it's fine to assume the user of this package should correctly initialise the struct before they use them. Let's think about it this way. Should we only export the interface for all the structs with a `NewThatStruct()` function? I think that could be overdesign.

Reply:
A drawback of using an interface here would be we need to port the methods we used from `lru.Cache` in the interface. This makes the struct embedding less useful.

Reply:
> IMO it's fine to assume the user of this package should correctly initialise the struct before they use them.

I am concerned with this since the internal values are not exported (correctly so), meaning if someone were to use the zero value, it would panic. 

I strongly suggest that this moves to be a non exported type and an interface is used in its place.

Reply:
The `cache` package is moved to `spanmetricsprocessor/internal`.